examples/game-of-life-challenge/src/contract.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg_attr(target_arch = "wasm32", no_main)]
+
+mod state;
+
+use async_graphql::ComplexObject;
+use gol_challenge::{GolChallengeAbi, Operation};
+use linera_sdk::{
+    linera_base_types::WithContractAbi,
+    views::{RootView, View},
+    Contract, ContractRuntime,
+};
+use serde::{Deserialize, Serialize};
+use state::{GolChallengeState, Solution};
+
+pub struct GolChallengeContract {
+    state: GolChallengeState,
+    runtime: ContractRuntime<Self>,
+}
+
+linera_sdk::contract!(GolChallengeContract);
+
+impl WithContractAbi for GolChallengeContract {
+    type Abi = GolChallengeAbi;
+}
+
+impl Contract for GolChallengeContract {
+    type Message = Message;
+    type InstantiationArgument = ();
+    type Parameters = ();
+    type EventValue = ();
+
+    async fn load(runtime: ContractRuntime<Self>) -> Self {
+        let state = GolChallengeState::load(runtime.root_view_storage_context())
+            .await
+            .expect("Failed to load state");
+        GolChallengeContract { state, runtime }
+    }
+
+    async fn instantiate(&mut self, _arg: ()) {
+        log::trace!("Instantiating");
+        self.runtime.application_parameters(); // Verifies that these are empty.
+    }
+
+    async fn execute_operation(&mut self, operation: Operation) {
+        log::trace!("Handling operation {:?}", operation);
+        let Operation::SubmitSolution {
```

Comment:
We could check solutions on the user chain, then send a message to claim the prize, for scalability.